### PR TITLE
Add snipe

### PR DESCRIPTION
**Pull request by mr-eyes**
This PR to add the package `snipe`, a massive-scale lightweight sequence assessment tool.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by mr-eyes** [comment]
@BiocondaBot  please add label

File: recipes/snipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+package:
+  name: {{ name|lower }}
+  version: "{{ version }}"
+
+source:
+  url: https://github.com/snipe-bio/snipe/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: a25432955f992119c8ce69e38fb03e7d79eff8c2880398944b61b0b628fd7d01
+
+
+entry_points:
```
Comment (martin-g) [inline recipes/snipe/meta.yaml:13]: This should be under `build` - https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#python-entry-points

**Review by martin-g — CHANGES_REQUESTED** [review]

File: recipes/snipe/meta.yaml (hunk 1 above)
Comment (mr-eyes) [reply recipes/snipe/meta.yaml]: Fixed; Thanks!

File: recipes/snipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - pytest
+    - numpy >=1.22,<2.0
+    - pathos
+    - pandas
+    - tqdm
+    - lzstring
+    - rapidfuzz
+    - zlib
+    - pyfastx >=2.1.0,<3.0a0
+    - sourmash >=4.8.11
```
[the remark this answers is not in the document]
Comment (martin-g) [reply recipes/snipe/meta.yaml:39]: @mr-eyes is the above applicable?

File: recipes/snipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  license_family: "AGPL"
+  license_file: "LICENSE.txt"
+
+extra:
+  recipe-maintainers:
+    - mr-eyes
+  authors:
+    - mr-eyes
+    - drtamermanosur
+  channels:
```
Comment (martin-g) [inline recipes/snipe/meta.yaml:58]: This is not needed. The channels are defined/hardcoded in Bioconda-utils package

Comment (mr-eyes) [reply recipes/snipe/meta.yaml]: resolved; thank you!

File: recipes/snipe/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mr-eyes) [reply recipes/snipe/meta.yaml]: yes, thank you!

**Review by martin-g — APPROVED** [review]

**Pull request merged**